doc/source/user/build_system/flags.rst

```diff
@@ -3,6 +3,182 @@
 Flags: constraints in dependencies
 ==================================
 
-.. todo::
+Flags in FuseSoC are global boolean variables that influence the dependency resolution and other aspects of the build.
+Flags can be used in multiples places in :term:`core files <core file>` to take conditional actions.
+For example, dependencies and files can be included depending on a flag, different toplevels can be selected, and much more.
 
-   Document flags.
+Facts about flags:
+
+* Flags are boolean variables, they are either "set" (true) or "unset" (false).
+* Flags are global, i.e. they apply to the whole build.
+  All cores see the same flags under the same name.
+* A valid flag name starts with a letter, and consists only of letters from the English alphabet, numbers, and the underscore (``_``).
```

Comment:
The naming rules are more strict than they need to be at the moment, but I thought it would be good to give us a bit of headroom for a different parser without restricting users too much -- these naming rules are already "common sense".

Reply:
Agree